core/src/main/java/com/valtech/aem/saas/core/common/resource/ParentResource.java

```diff
@@ -0,0 +1,27 @@
+package com.valtech.aem.saas.core.common.resource;
+
+import java.util.Optional;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import org.apache.sling.api.resource.Resource;
+
+@RequiredArgsConstructor
+public final class ParentResource {
+
+  @NonNull
+  private final Resource resource;
+
+  public Optional<Resource> getParentWithResourceType(String resourceType) {
+    return Optional.ofNullable(getParentWithResourceType(resource, resourceType));
+  }
+
+  private Resource getParentWithResourceType(@NonNull Resource resource, String resourceType) {
+    if (resource.getParent() != null) {
+      if (resource.getParent().isResourceType(resourceType)) {
+        return resource.getParent();
+      }
+      return getParentWithResourceType(resource.getParent(), resourceType);
```

Comment:
this should not happen due to the check on line 19

Reply:
In line 23 you pass in resource.getParent(), and let's assume it would be null, then on line 19 directly resource.getParent() is called again without a check if resource is not null.

Reply:
My bad, was blind...